File: medical-appointment/src/main/java/com/medical/appointment/model/entity/Appointment.java
package com.medical.appointment.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

/**
 * 预约实体类
 */
@Data
@TableName("appointment")
public class Appointment {
    /**
     * ID
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 病人ID
     */
    private String patientId;

    /**
     * 医生ID
     */
    private String doctorId;

    /**
     * 预约日期
     */
    private LocalDate date;

    /**
     * 开始时间
     */
    private LocalTime startTime;

    /**
     * 结束时间
     */
    private LocalTime endTime;

    /**
     * 备注
     */
    private String remark;

    /**
     * 状态：normal-正常，to_be_rescheduled-待重新安排，cancelled-已取消
     */
    private String status;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
} 

File: web-dashboard/src/types/appointment.ts
export interface RawAppointment {
  id: string;
  patient: {
    id: string;
    nickname: string;
    avatarUrl?: string;
  };
  doctor: {
    id: string;
    firstName: string;
    lastName: string;
    avatarUrl?: string;
  };
  startAt: string;
  endAt: string;
  remark: string;
  status: "normal" | "to_be_rescheduled" | "cancelled";
  createdAt: string;
}
